packages/nimble-components/src/table-column/number-text/models/number-text-unit-format.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    UnitFormat,
+    type UnitFormatOptions
+} from '../../../utilities/unit-format/unit-format';
+import { passthroughUnitScale } from '../../../utilities/unit-format/unit-scale/passthrough-unit-scale';
+import { NumberTextFormat } from '../types';
+import { DefaultUnitFormat } from '../../../utilities/unit-format/default-unit-format';
+import { DecimalUnitFormat } from '../../../utilities/unit-format/decimal-unit-format';
+
+export interface NumberTextUnitFormatOptions extends UnitFormatOptions {
+    numberTextFormat?: NumberTextFormat;
+    decimalDigits?: number;
+    decimalMaximumDigits?: number;
+}
+type ResolvedNumberTextUnitFormatOptions = NumberTextUnitFormatOptions &
+Required<UnitFormatOptions>;
+
+/**
+ * Format for numbers (with optional units) in a number-text table column.
+ */
+export class NumberTextUnitFormat extends UnitFormat {
+    private static readonly defaultDecimalDigits = 2;
+    private readonly resolvedUnitFormat: DefaultUnitFormat | DecimalUnitFormat;
+    private readonly _resolvedOptions: ResolvedNumberTextUnitFormatOptions;
+
+    public constructor(locale: string, options?: NumberTextUnitFormatOptions) {
+        super();
+        this._resolvedOptions = this.resolveOptions(options);
+        this.resolvedUnitFormat = this.resolveUnitFormat(
+            locale,
+            this._resolvedOptions
+        );
+    }
+
+    public override resolvedOptions(): ResolvedNumberTextUnitFormatOptions {
+        return { ...this._resolvedOptions };
+    }
+
+    public optionsMatch(targetOptions?: NumberTextUnitFormatOptions): boolean {
+        const targetResolvedOptions = this.resolveOptions(targetOptions);
+        return (
+            this._resolvedOptions.decimalDigits
+                === targetResolvedOptions.decimalDigits
+            && this._resolvedOptions.decimalMaximumDigits
+                === targetResolvedOptions.decimalMaximumDigits
+            && this._resolvedOptions.numberTextFormat
+                === targetResolvedOptions.numberTextFormat
+            && this._resolvedOptions.unitScale === targetResolvedOptions.unitScale
+        );
+    }
```

Comment:
Can you add some test cases for this optionsMatch behavior. A unique bit is that it is comparing resolved options, so we should exercise some resolved option cases, particularly around defaults.

Some example option sets to compare in optionsMatch
- undefined optionsMatch {}
- {} optionsMatch {numberTextFormat: default, decimalDigits: 7}
- {numberTextFormat: decimal} optionsMatch {numberTextFormat: decimal, decimalDigits: 2}